aemedge/scripts/scripts.js

```diff
@@ -120,8 +145,70 @@ export async function decorateMain(main, shouldDecorateTemplates = true) {
   decorateSections(main);
   decorateBlocks(main);
   decorateMultiColumnSections(main);
+  decorateFragments(main);
 }
 
+/**
+ * Loads a fragment.
+ * @param {string} path The path to the fragment
+ * @returns {HTMLElement} The root element of the fragment
+ */
+export async function loadFragment(path) {
+  if (path && path.startsWith('/')) {
+    const resp = await fetch(`${path}.plain.html`);
+    if (resp.ok) {
+      const main = document.createElement('main');
+      main.innerHTML = await resp.text();
+
+      // reset base path for media to fragment base
+      const resetAttributeBase = (tag, attr) => {
+        main.querySelectorAll(`${tag}[${attr}^="./media_"]`).forEach((elem) => {
+          elem[attr] = new URL(elem.getAttribute(attr), new URL(path, window.location)).href;
+        });
+      };
+      resetAttributeBase('img', 'src');
+      resetAttributeBase('source', 'srcset');
+
+      await decorateMain(main, false);
+      await loadBlocks(main);
+      return main;
+    }
+  }
+  return null;
+}
+
+/**
+ * Replace a link placeholder with a corresponding fragment, loaded asynchronously.
+ * @param {Element} link The link placeholder, which has the URL in its innerHTML
+ */
+async function replaceLinkPlaceHolderWithFragment(link) {
+  const path = link.innerHTML;
+  const fragment = await loadFragment(path);
+  if (fragment) {
+    const fragmentSection = fragment.querySelector(':scope .section');
+    if (fragmentSection) {
+      link.closest('.section').classList.add(...fragmentSection.classList);
+      let nodeToReplace = link;
+      while (nodeToReplace.parentNode.tagName !== 'DIV' && nodeToReplace.childNodes.length === 1) {
```

Comment:
Why the parent of placeholder link cannot be a `<div>` ?

Reply:
It can be, but we shouldn't replace that div since divs are structural. So we go up the parent chain only until we don't hit a div.